analysis/filter/MergeFilter/src/main/kotlin/de/maibornwolff/codecharta/filter/mergefilter/ProjectMerger.kt

```diff
@@ -0,0 +1,40 @@
+package de.maibornwolff.codecharta.filter.mergefilter
+
+import de.maibornwolff.codecharta.model.Node
+import de.maibornwolff.codecharta.model.Project
+
+class ProjectMerger(val projects: List<Project>) {
+
+    fun extractProjectName(): String {
+        val projectNames = projects.map { p -> p.projectName }.toSortedSet()
+        when (projectNames.size) {
+            1 -> return projectNames.first()
+            else -> throw MergeException("Projects contain several project names : " + projectNames)
+        }
+    }
+
+    fun extractApiVersion(): String {
```

Comment:
Does not only extract the API version but also checks for consistency between the two projects -> maybe split this into two separate functions.